R/history.R

```diff
@@ -0,0 +1,16 @@
+#' Update URL in browser's location bar
+#'
+#' This function updates the client browser's query string in the location bar.
+#' It typically is called from an observer. Note that this will not work in
+#' Internet Explorer 9 and below.
+#'
+#' @param state State to save on the history stack.
+#' @param title Not currently in use.
+#' @param url New url to display on the window.
+#' @param session A Shiny session object.
+#' @seealso \code{\link{enableBookmarking}} for examples.
+#' @export
+pushState <- function(state, title, url,
```

Comment:
I can't decide whether I like how closely this follows the HTML5 API. With `updateQueryString` we don't assume you know the HTML5 jargon. For sure, `pushState` is far too general a function name--keep in mind that in HTML5 you're accessing it via the history object, i.e. `history.pushState()` is far less vague than just `pushState()`.

Maybe `navigateQueryString(queryString, session)`?